node/pkg/vaa/structs_test.go

```diff
@@ -207,6 +207,42 @@ func TestHexDigest(t *testing.T) {
 	assert.Equal(t, vaa.HexDigest(), expected)
 }
 
+func TestMarshal(t *testing.T) {
+	expectedBytes := []byte{0x1, 0x0, 0x0, 0x0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0x0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0x20, 0x61, 0x61, 0x61, 0x61, 0x61, 0x61}
+	vaa := getVaa()
+	marshalBytes, err := vaa.Marshal()
+	assert.Nil(t, err)
+	assert.Equal(t, expectedBytes, marshalBytes)
+}
+
+func TestUnmarshal(t *testing.T) {
+	vaaBytes := []byte{0x1, 0x0, 0x0, 0x0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0x0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0x20, 0x61, 0x61, 0x61, 0x61, 0x61, 0x61}
+	vaa1 := getVaa()
+	vaa2, err := Unmarshal(vaaBytes)
+	assert.Nil(t, err)
+	assert.Equal(t, &vaa1, vaa2)
+}
+
+func TestUnmarshalTooBig(t *testing.T) {
+	vaa := getVaa()
+
+	// Overwrite an oversized payload for the VAA that we cannot unmarshal
+	var payload []byte
+	for i := 0; i < 2000; i++ {
+		payload = append(payload, 'a')
+	}
+	vaa.Payload = payload
+
+	// Let's marshal the VAA to bytes to unmarshaled
+	marshalBytes, err := vaa.Marshal()
+	assert.Nil(t, err)
+
+	// Let's now unmarshal the oversized VAA and cause it to panic
+	vaa2, err2 := Unmarshal(marshalBytes)
+	assert.Nil(t, vaa2)
+	assert.NotNil(t, err2)
```

Comment:
This should fail, seeking feedback on the desired behavior before I adjust the implementation.